tests/samplers_tests/test_nsgaii.py

```diff
@@ -146,6 +148,151 @@ def constraints_func(trial: FrozenTrial) -> Sequence[float]:
         assert trial.system_attrs[_CONSTRAINTS_KEY] == (trial.number,)
 
 
+@pytest.mark.parametrize("direction1", [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE])
+@pytest.mark.parametrize("direction2", [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE])
+@pytest.mark.parametrize(
+    "constraints_list",
+    [
+        [[]],  # empty constraint
+        [[-float("inf")], [-1], [0]],  # 1d constraints
+        [
+            [c1, c2] for c1 in [-float("inf"), -1, 0] for c2 in [-float("inf"), -1, 0]
+        ],  # 2d-constraints
+    ],
```

Comment:
@c-bata mentioned that what `constraints_list` may not be clear.

Reply:
I didn't understand why the dimensions of these constraints are changed like 1d-constraints and 2d-constraints even though `constraints_fn` returns a sequence of floats.

However, now I notice that all of these constraints are actually 2-dimensional list and the only difference is the size of the arrays.`single constraint` and `multiple constraints` might be more appropriate than `1d constraints` and `2d constraints`.